pkg/server/sotw/v3/server.go

```diff
@@ -407,6 +407,8 @@ func (s *server) StreamHandler(stream Stream, typeURL string) error {
 			}
 			select {
 			case reqCh <- req:
+			case <-stream.Context().Done():
+				return
```

Comment:
I wonder if this can replace the atomic + TODO below, which seems like an incomplete workaround for this same problem.  Not sure if there are regression tests specifically for that atomic, but if so we could remove it and see if they still passed.

Reply:
Unfortunately it seems like the tests pass on main without this logic. I'm not quite sure what this is trying solve, if this is making sure that we don't try to send to `reqCh` after the consuming loop has ended I would think selecting on the stream context when sending would cover our bases, as I assume the stream context is canceled once `StreamHandler` exists? 

The TODO seems to talk about sending to a closed channel though, but this doesn't make sense to me since this goroutine is responsible for closing this channel?

Reply:
I think the original code was referring to the race that `Recv()` unblocks with no error right when `process()` exits. `reqCh` is unbuffered so it might block without draining. With this change, once the handler exits, we should in theory receive stream context done, so the blocked channel write will unblock and exit. Thanks for the fix.